Dockerfile

```diff
@@ -1,16 +1,33 @@
-FROM ubuntu:22.04
+FROM ubuntu:22.04 AS builder
 
-WORKDIR /metachain 
+RUN apt update && apt upgrade -y
+RUN apt install -y curl
 
-# Requires copy the binary to `build` folder beforehand
-COPY build/* /metachain 
+WORKDIR /metachain
+RUN curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain none --profile minimal -y
+
+ENV PATH=$PATH:/root/.cargo/bin
+RUN apt install -y cmake clang
+
+COPY rust-toolchain.toml Cargo.toml Cargo.lock ./
+# TODO: docker layer cache, blocked by https://github.com/rust-lang/cargo/issues/2644
+# RUN cargo fetch
+
+COPY meta ./meta
+ARG PROFILE=release
+ENV PROFILE ${PROFILE}
+RUN cargo build --$PROFILE --all
+
+
+FROM ubuntu:22.04 AS runner
+WORKDIR /metachain
+COPY --from=builder /metachain/target/release/meta-node .
 
 # 30333 for p2p traffic
 # 9933 for RPC call
 # 9944 for Websocket
 # 9615 for Prometheus (metrics)
-EXPOSE 30333 9933 9944
+EXPOSE 30333 9933 9944 9615
```

Comment:
```suggestion
EXPOSE 30333 9333 9944 9615 39333 19933 19944  # from https://github.com/DeFiCh/metachain/blob/d3f4a9b36eb25d7340a8b138795882cada7c60e5/packages/network/src/NetworkConfig.ts
```

Reply:
https://github.com/DeFiCh/metachain/pull/163#discussion_r1030038430

hmm.. adding `39333 19933 19944` is not needed btw...
the `EXPOSE` does not make the ports of the container accessible to the host... `EXPOSE` just makes it available only for inter-container communication

Reply:
Hmm

I think that's the case; it's probably needed if the nodes are setup to use inter-container communication. Since we're moving to inter-container testing instead of host test further down the road.